### ISSUE TITLE
Driver Not Found

### ISSUE THREAD

**Issue by cheldernunes** [debezium/debezium-examples#11]
**This log show the error:**
`connect_1    | org.apache.kafka.connect.errors.ConnectException: java.sql.SQLException: No suitable driver found for jdbc:postgresql://postgres:5432/inventory?user=postgresuser&password=postgrespw`

**Curl Request about status of connector**
`curl -s -X GET http://localhost:8083/connectors/jdbc-sink/status`
**return**
`{"name":"jdbc-sink","connector":{"state":"RUNNING","worker_id":"172.18.0.6:8083"},"tasks":[{"state":"FAILED","trace":"org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:484)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:265)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:182)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:150)\n\tat org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:146)\n\tat org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:190)\n\tat java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n","id":0,"worker_id":"172.18.0.6:8083"}]
`

**Comment by gunnarmorling** [comment]
How are you starting the Kafka Connect server? Have you deployed the
Postgres JDBC driver? You can take a look at how it can be done using this
Docker file:


https://github.com/debezium/debezium-examples/blob/master/unwrap-smt/debezium-jdbc/Dockerfile

Which is used in the Compose file of the SMT example:


https://github.com/debezium/debezium-examples/blob/master/unwrap-smt/docker-compose.yaml#L34

**Comment by cheldernunes** [comment]
Hi, thanks for the reply.
I executed exactly this docker project, according to the readme help.
Everything initializes correctly (zookeeper, kafka,kafka-connect, mysql, pgsql). But when I start the connector (@source.json) this problem occurs.

**Comment by gunnarmorling** [comment]
> But when I start the connector (@source.json) this problem occurs.

Above you said it's about the sink connector. I'm confused :)

**Comment by cheldernunes** [comment]
Yes. is about himself!
The sink runs correctly until I run the source. I'll give you the status of execution to understand
```
$ composer-docker run
```
All services run (OK)
```
$curl -i -X POST -H "Accept:application/json" -H  "Content-Type:application/json" http://localhost:8083/connectors/ -d @jdbc-sink.json
````
HTTP/1.1 201 Created (OK)
```
$curl -s -X GET http://localhost:8083/connectors/jdbc-sink/status
{"name":"jdbc-sink","connector":{"state":"RUNNING","worker_id":"172.18.0.6:8083"},"tasks":[{"state":"RUNNING","id":0,"worker_id":"172.18.0.6:8083"}]}
```
Checked state of sink(OK)
```
$curl -i -X POST -H "Accept:application/json" -H  "Content-Type:application/json" http://localhost:8083/connectors/ -d @source.json
````
HTTP/1.1 201 Created(OK)
```
$curl -s -X GET http://localhost:8083/connectors/inventory-connector/status
{"name":"inventory-connector","connector":{"state":"RUNNING","worker_id":"172.18.0.6:8083"},"tasks":[{"state":"RUNNING","id":0,"worker_id":"172.18.0.6:8083"}]}
```
Checked state of source (OK)

Now in to LOG(ERROR)
```
connect_1    | Caused by: java.sql.SQLException: No suitable driver found for jdbc:postgresql://postgres:5432/inventory?user=postgresuser&password=postgrespw
```
and  now Re-check Shink show me (ERROR)
```
curl -s -X GET http://localhost:8083/connectors/jdbc-sink/status
{"name":"jdbc-sink","connector":{"state":"RUNNING","worker_id":"172.18.0.6:8083"},"tasks":[{"state":"FAILED","trace":"org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:484)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:265)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:182)\n\tat org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:150)\n\tat org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:146)\n\tat org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:190)\n\tat java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n","id":0,"worker_id":"172.18.0.6:8083"}]}
```

END

**Comment by jpechane** [comment]
Have not you hit https://issues.apache.org/jira/browse/KAFKA-5953?

**Comment by gunnarmorling** [comment]
Hum, that's really weird. I don't think KAFKA-5953 should show up here, as in this project we add the Postgres driver straight to _/kafka/lib_.

@cheldernunes, Can you double check whether you actually run the `debezium/connect-jdbc` image (produced by this example project) and not the regular `debezium/connect` image (e.g. by running `docker ps`)? Could you also check whether the PG driver JAR is at the right place:

    docker-compose exec connect bash -c 'ls /kafka/libs/*postgresql*'

It should print "/kafka/libs/postgresql-42.1.4.jar". If that's actually the case, I'm out of ideas why it wouldn't work (and indeed I can't reproduce the issue following the steps given above).

**Comment by cheldernunes** [comment]
It's a live !
KAFKA-5953 says that error was fixed in version 1.0.0. So I changed debezium/kakfa image.

[https://github.com/debezium/docker-images/blob/master/kafka/0.6/Dockerfile#L9](https://github.com/debezium/docker-images/blob/master/kafka/0.6/Dockerfile#L9) to `1.0.0`
[https://github.com/debezium/docker-images/blob/master/kafka/0.6/Dockerfile#L15](https://github.com/debezium/docker-images/blob/master/kafka/0.6/Dockerfile#L15) to  `SHA512HASH=B698028485D460C494AF942B2DFA41C7342A2AD6052B543C95289CD1E832A1EBCF679B7E568DCABC342C799765337B94F91BC0CF0EE915534CDF82D1635A622A`
and
[https://github.com/debezium/docker-images/blob/master/kafka/0.6/Dockerfile#L41](https://github.com/debezium/docker-images/blob/master/kafka/0.6/Dockerfile#L41) to
`RUN echo "$SHA512HASH  /tmp/kafka.tgz" | sha512sum -c - &&\`
I removed all containers previous downloaded and I generated new images build. Then it worked here. 

Thank you @gunnarmorling  @jpechane
